pkg/aggregator/no_aggregation_stream_worker.go

```diff
@@ -135,13 +148,24 @@ func (w *noAggregationStreamWorker) run() {
 						n := time.Now()
 						if serializedSamples > 0 && lastStream.Before(n.Add(-time.Second*1)) {
 							log.Debug("noAggregationStreamWorker: triggering an automatic payloads flush to the forwarder (no traffic since 1s)")
+							tlmNoAggFlush.Add(1)
 							break mainloop // end `Serialize` call and trigger a flush to the forwarder
 						}
 
 					// receiving samples
 					case samples := <-w.samplesChan:
 						log.Debugf("Streaming %d metrics from the no-aggregation pipeline", len(samples))
+						var processed int
+
 						for _, sample := range samples {
+							mtype, supported := metricSampleAPIType(sample)
+
+							if !supported {
+								log.Warnf("Discarding unsupported metric sample in the no-aggregation pipeline for sample '%s', sample type '%s'", sample.Name, sample.Mtype.String())
```

Comment:
This has potential to produce way too many log messages. Would it make sense to throttle them a bit? Once per metric per minute or something like that?

Reply:
Yeah I don't know, this would mean somebody is specifically crafting messages with a wrong type (the official clients won't be able to send unsupported samples to the server) but I agree with you it could be verbose...
I can consider a basic throttling stopping this message after a thousand of them for instance (not per metric though to avoid map accesses etc...). I'll see what throttling logging utility we have available.

Reply:
> Yeah I don't know, this would mean somebody is specifically crafting messages with a wrong type

Not necessarily, it could also be caused a bug in the client, or by using a newer client with an older agent.

Reply:
> it could also be caused a bug in the client, or by using a newer client with an older agent.

You are right, I'll implement a limit.

Reply:
Addressed here in this commit with the introduction of a very basic `SimpleThrottler`: https://github.com/DataDog/datadog-agent/pull/13427/commits/40e1cfc12caa5f04a3fc14ce84a31ed01cf50a95

Reply:
It has been through many feedback from ASC, better to review the whole code change of the PR now... Sorry for this.